src/accordion/accordion.spec.ts

```diff
@@ -27,304 +28,346 @@ function expectOpenPanels(nativeEl: HTMLElement, openPanelsDef: boolean[]) {
 }
 
 describe('ngb-accordion', () => {
-  let html: string;
-
-  beforeEach(() => {
-    html = `
-      <ngb-accordion #acc="ngbAccordion" [closeOthers]="closeOthers" [activeIds]="activeIds" 
-        (change)="changeCallback($event)" [type]="classType">
-        <ngb-panel *ngFor="let panel of panels" [id]="panel.id" [disabled]="panel.disabled" [type]="panel.type">
-          <template ngbPanelTitle>{{panel.title}}</template>       
-          <template ngbPanelContent>{{panel.content}}</template>        
-        </ngb-panel>        
-      </ngb-accordion>
-      <button *ngFor="let panel of panels" (click)="acc.toggle(panel.id)">Toggle the panel {{ panel.id }}</button>
-    `;
-  });
-
-  it('should have no open panels', async(inject([TestComponentBuilder], (tcb) => {
-       tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
-         fixture.detectChanges();
-         expectOpenPanels(fixture.nativeElement, [false, false, false]);
-       });
-     })));
-
-  it('should toggle panels based on "activeIds" values', async(inject([TestComponentBuilder], (tcb) => {
-       tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
-         const tc = fixture.componentInstance;
-         const el = fixture.nativeElement;
-
-         // as array
-         tc.activeIds = ['one', 'two'];
-         fixture.detectChanges();
-         expectOpenPanels(el, [true, true, false]);
-
-         tc.activeIds = ['two', 'three'];
-         fixture.detectChanges();
-         expectOpenPanels(el, [false, true, true]);
+  let html = `
+    <ngb-accordion #acc="ngbAccordion" [closeOthers]="closeOthers" [activeIds]="activeIds"
+      (change)="changeCallback($event)" [type]="classType">
+      <ngb-panel *ngFor="let panel of panels" [id]="panel.id" [disabled]="panel.disabled" [type]="panel.type">
+        <template ngbPanelTitle>{{panel.title}}</template>
+        <template ngbPanelContent>{{panel.content}}</template>
+      </ngb-panel>
+    </ngb-accordion>
+    <button *ngFor="let panel of panels" (click)="acc.toggle(panel.id)">Toggle the panel {{ panel.id }}</button>
+  `;
+
+  beforeEach(() => { TestBed.configureTestingModule({declarations: [TestComponent], imports: [NgbAccordionModule]}); });
+
+  it('should have no open panels', async(() => {
+       TestBed.overrideComponent(TestComponent, {set: {template: html}});
```

Comment:
In this particular test suite a template is shared by many (all?) test cases so I think that we could have the `TestBed.overrideComponent(TestComponent, {set: {template: html}});` in the `beforeEach` block only and not repeat it in all the `it` blocks.